lib/messageformat.js

```diff
@@ -133,6 +133,36 @@ MessageFormat.formatters = {
   }
 };
 
+/** Predefined number formatting function enforcing Bidi Structured Text by using UCC
+ * Prevents intermingling of text contained in placeholders with the rest of message text.
+ * Maintains the integrity of Bidi text flowing according to directionality of language script.
+ * For reference to Bidi structured text see: http://cldr.unicode.org/development/development-process/design-proposals/bidi-handling-of-structured-text
+ *  @param {string} - Placeholder text content
+ *  @param {string} - Locale
+ *  @returns {string} - The resultant text fixed by UCC Bidi strong marks
+ *  @examples: (upper case stands for Bidi (Arabic/Hebrew) characters, output is shown as displayed in browser page
+ *  > var MessageFormat = require('messageformat');
+ *  > var mf = new MessageFormat('en', null, {"bidiStructuredText": MessageFormat.formatters.bidiStructuredText} );
+ *  > var mfunc = mf.compile("{0} >> {1} >> {2}");
+ *  > mfunc([ "first_english_word", "SECOND_ARABIC_WORD", "THIRD_ARABIC_WORD" ])
+ * structured output:
+ *     "first_english_word >> SECOND_ARABIC_WORD  >> THIRD_ARABIC_WORD"
+ * distorted output when MessageFormat gets initialized with no structured text formatter:
+ *     "first_english_word >> THIRD_ARABIC_WORD  << SECOND_ARABIC_WORD"
+ *  */
+MessageFormat.formatters.bidiStructuredText = function( text, locale ) { //shensis mesage_format
+  var isLocaleRTL = function(locale) {
+    var rtlLanguages = ["ar", "fa", "he", "ks-Arab", 'pa-Arab', "ps",  "ur", "uz-Arab", "ks"];
+    return new RegExp("^" + rtlLanguages.join( "|^" ) + "$").test( locale );
+  };
```

Comment:
Why does the RegExp get a single `$` at the end, verifying that only the last language in the array has its test anchored at the end? As is, `isLocaleRTL('fax')` is `true` while `isLocaleRTL('ksx')` is `false`, for no discernible reason.